File: src/priv/bajdcc/util/lexer/regex/IRegexStringFilterHost.java
package priv.bajdcc.util.lexer.regex;

import priv.bajdcc.util.lexer.algorithm.ITokenAlgorithm;

/**
 * 字符串过滤主体
 * 
 * @author bajdcc
 *
 */
public interface IRegexStringFilterHost {
	/**
	 * 设置字符转换算法
	 * @param alg 字符转换算法
	 */
	public void setFilter(ITokenAlgorithm alg);
}

File: src/main/kotlin/com/bajdcc/util/lexer/regex/IRegexStringFilterHost.kt
package com.bajdcc.util.lexer.regex

import com.bajdcc.util.lexer.algorithm.ITokenAlgorithm

/**
 * 字符串过滤主体
 *
 * @author bajdcc
 */
interface IRegexStringFilterHost {
    /**
     * 设置字符转换算法
     *
     * @param alg 字符转换算法
     */
    fun setFilter(alg: ITokenAlgorithm)
}
